src/WopiTokenManager.php

```diff
@@ -0,0 +1,163 @@
+   *
+   * @param string $token
+   *   The token to verify.
+   *
+   * @return array|null
+   *   Data decoded from the token, or NULL on failure or if the token has
+   *   expired.
+   *
+   * @throws \Drupal\collabora_online\Exception\CollaboraNotAvailableException
+   *   The key to use by Collabora is empty or not configured.
+   */
+  public function decodeAndVerify(
+    #[\SensitiveParameter]
+    string $token,
+  ): array|null {
+    $key = $this->getKey();
+    try {
+      $payload = (array) JWT::decode($token, new Key($key, 'HS256'));
+    }
+    catch (\Exception $e) {
+      $this->logger->error($e->getMessage());
+      return NULL;
+    }
+    if (!isset($payload['exp'])) {
+      return NULL;
+    }
+    if ($payload['exp'] < gettimeofday(TRUE)) {
+      // Token is expired.
+      return NULL;
+    }
+    return $payload;
+  }
+
+  /**
+   * Creates a JWT token for a media entity.
+   *
+   * The token will carry the following:
+   *
+   * - fid: the Media id in Drupal.
+   * - uid: the User id for the token. Permissions should be checked
+   *   whenever.
+   * - exp: the expiration time of the token.
+   * - wri: if true, then this token has write permissions.
+   *
+   * The signing key is stored in Drupal key management.
+   *
+   * @param array $payload
+   *   Values to encode in the token.
+   * @param int|float|null $expire_timestamp_reference
+   *   Reference to be filled with the expiration timestamp, in seconds.
+   *
+   * @return string
+   *   The access token.
+   *
+   * @param-out int|float $expire_timestamp_reference
+   *
+   * @throws \Drupal\collabora_online\Exception\CollaboraNotAvailableException
+   *   The key to use by Collabora is empty or not configured.
+   */
+  public function encode(array $payload, int|float|null &$expire_timestamp_reference = NULL): string {
+    $expire_timestamp_reference = $this->getExpireTimestamp();
+    $payload['exp'] = $expire_timestamp_reference;
+    $key = $this->getKey();
+    $jwt = JWT::encode($payload, $key, 'HS256');
+
+    return $jwt;
+  }
+
+  /**
+   * Gets a token expiration timestamp based on the configured TTL.
+   *
+   * @return float
+   *   Expiration timestamp in seconds, with millisecond accuracy.
+   */
+  protected function getExpireTimestamp(): float {
+    $cool_settings = $this->configFactory->get('collabora_online.settings')->get('cool');
+    $ttl_seconds = $cool_settings['access_token_ttl'] ?? 0;
+    // Set a fallback of 24 hours.
+    $ttl_seconds = $ttl_seconds ?: 86400;
+
+    return gettimeofday(TRUE) + $ttl_seconds;
+  }
+
+  /**
+   * Obtains the signing key from the key storage.
+   *
+   * @return string
+   *   The key value.
+   *
+   * @throws \Drupal\collabora_online\Exception\CollaboraNotAvailableException
+   *   The key to use by Collabora is empty or not configured.
+   */
+  protected function getKey(): string {
+    $default_config = $this->configFactory->get('collabora_online.settings');
```

Comment:
Yeah, I was wondering if we should introduce a dedicated service for this configuration.
I am hesitating because I don't find this elsewhere in Drupal.

Imo it is ok to do this kind of local refactoring first, and later we can unify the config handling.